quinn-proto/src/connection/streams/recv.rs

```diff
@@ -29,6 +30,15 @@ impl Recv {
         })
     }
 
+    /// Reset to the initial state
+    pub(super) fn reinit(&mut self, initial_max_data: u64) {
+        self.state = RecvState::default();
+        self.assembler.reinit();
+        self.sent_max_stream_data = initial_max_data;
+        self.end = 0;
+        self.stopped = false;
```

Comment:
As above.

Reply:
If I do this, in the generated code in --release mode I do see the old value being copied on the stack, and then assembler copied back into self. Since this is in a relatively hot path for us, I'd rather not have that. But also given that we are in micro-opt territory, happy to do it if you feel strongly about it.

Reply:
Maybe a slightly different construct would work better? e.g.:
```rs
let old_data = mem:;take(&mut self.data);
*self = Self::default();
self.data = old_data;
self.data.clear();
```

Reply:
for Assembler::reinit, this works. For Recv::reinit, we still have the same issue that it copies Assembler twice, and since Assembler is relatively large (88 bytes), that takes a few insns.

Reply:
Oh well, I don't feel that strongly. Maybe leave a comment inside the struct definition as a reminder to update `reinit` when adding or modifying fields.

Reply:
done